src/bulls_and_cows_lib/game.cpp

```diff
@@ -9,52 +9,118 @@
+            //std::cout << "Bulls : " << newattemp.feedback.bulls << " Cows : " << newattemp.feedback.cows<<"\n";
+            myavcboard.attempts_and_feedbacks.push_back(newattemp);
+            std::cout << "Secret code is : " << myavcboard.secret_code.value << "\n";
+            if (bulls_and_cows::is_win(game_options, myavcboard))break;
+            bulls_and_cows::remove_incompatible_codes_from_possible_solutions(newattemp, myavc);
+            Sleep(2000);
+        }
     }
 
-    void configure_game_options(GameOptions& game_options)
+    void configure_game_options(GameOptions & game_options)
     {
-        std::cout << "TODO:\n"
-                     "    DO\n"
-                     "       Display the current game options\n"
-                     "       Display the game options menu\n"
-                     "       Ask the user to type its choice\n"
-                     "       Treat the user's choice\n"
-                     "    UNTIL user's choice is to go back to main menu\n";
+        std::ifstream in{"game_options.txt"};
+
+        bool get_out = false;
+        while (!get_out)
+        {
+            display_game_options(std::cout, game_options);
+            display_game_options_menu(std::cout);
+            std::cin.clear();
+            const auto user_choice = ask_game_options_menu_choice(std::cin);
+
+                switch (user_choice)
+                {
+                case GameOptionsMenuChoice::BackToMain:
+                    get_out = true;
+                    break;
+                case GameOptionsMenuChoice::ModifyMaximumAllowedCharacter:
+                    std::cout << "Enter the maximum character you want:";
+                    game_options.maximum_allowed_character =
+                        ask_char_or_default(std::cin, game_options.maximum_allowed_character);
+                    break;
+                case GameOptionsMenuChoice::ModifyMaximumNumberOfAttempts:
+                    std::cout << "Enter the number of attempts you want:";
+                    game_options.max_number_of_attempts =
+                        ask_uint_or_default(std::cin, game_options.max_number_of_attempts);
+                    break;
+                case GameOptionsMenuChoice::ModifyMinimumAllowedCharacter:
+                    std::cout << "Enter the minimum character you want:";
+                    game_options.minimum_allowed_character =
+                        ask_char_or_default(std::cin, game_options.minimum_allowed_character);
+                    break;
+                case GameOptionsMenuChoice::ModifyNumberOfCharactersPerCode:
+                    std::cout << "Enter the number of character per code:";
+                    game_options.number_of_characters_per_code =
+                        ask_uint_or_default(std::cin, game_options.number_of_characters_per_code);
+                    break;
+                case GameOptionsMenuChoice::LoadOptions:
```

Comment:
C'est ici qu'il faudrait définir/déclarer la variable in.
Mais il faudrait mettre tout ce qu'il y a après les case dans un bloc d'accolades, sinon ça fait comme si la variable avait été déclarée avant le switch.

Reply:
Done .